gbpservice/nfp/lifecycle_manager/drivers/lifecycle_driver_base.py

```diff
@@ -0,0 +1,440 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from oslo_log import log as logging
+
+from gbpservice.nfp._i18n import _
+from gbpservice.nfp.lifecycle_manager.openstack import (
+    openstack_driver
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class LifeCycleDriverBase(object):
+    """Generic Driver class for Lifecycle handling of virtual appliances
+
+    Does not support sharing of virtual appliance for different chains
+    Does not support hotplugging interface to devices
+    Launches the VM with all the management and data ports and a new VM
+    is launched for each Network Service Instance
+    """
+    def __init__(self, supports_device_sharing=False, supports_hotplug=False,
+                 max_interfaces=5):
+        self.service_vendor = 'general'
+        self.supports_device_sharing = supports_device_sharing
+        self.supports_hotplug = supports_hotplug
+        self.maximum_interfaces = max_interfaces
+
+        # TODO[Magesh]: Try to move the following handlers to
```

Comment:
This code should only have compute_driver, networking_driver and not know about nova, neutron, gbp. All of this should be abstracted with the common interface using different drivers under lifecycle_manager/infra. Is this possible?